### PR TITLE
Chore: merge master into develop

### PR DESCRIPTION
**Pull request by pavitthrap**
This merges the updates (up to latest epoch 2.4 release) in `master` back to `develop`.

### CODE REVIEW

**Review by kantai — COMMENTED** [review]
This looks like it needs some merge fixes / artifact fixes -- the test build is currently failing on the branch.

**Comment by obycode** [comment]
In reviewing this, I noticed that there is nothing added to the changelog about the atlas changes. @jbencin please add something in a separate PR after this one is merged.

**Comment by obycode** [comment]
Looks like we are similarly missing a changelog entry for the named bitcoin wallet fix (https://github.com/stacks-network/stacks-blockchain/commit/89e0bcaf5e923746d1332248d59c752e6400aae0). @wileyj are you familiar enough with that one to add a changleog entry?

**Comment by wileyj** [comment]
> Looks like we are similarly missing a changelog entry for the named bitcoin wallet fix ([89e0bca](https://github.com/stacks-network/stacks-blockchain/commit/89e0bcaf5e923746d1332248d59c752e6400aae0)). @wileyj are you familiar enough with that one to add a changleog entry?

outside of reading the commit, no - but it does look eerily similar to the changes in #3694 . 
I can add a changelog though, the code/commit message is clear enough to do that

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ and this project adheres to the versioning scheme outlined in the [README.md](RE
   runtime error and include details about the error in the `vm_error` field of
   the receipt. Fixes issues #3154, #3328.
 
+- Added config setting `burnchain.wallet_name` which addresses blank wallets no 
+  longer being created by default in recent bitcoin versions. Fixes issue ##3596
```
Comment (obycode) [inline CHANGELOG.md:25]:
```suggestion
  longer being created by default in recent bitcoin versions. Fixes issue #3596
```

Comment (wileyj) [reply CHANGELOG.md]: good catch! updated

Comment (obycode) [reply CHANGELOG.md]: Thanks!

**Review by wileyj — APPROVED** [review]
shipit

**Pull request merged**